src/lxc/lxc_attach.c

```diff
@@ -273,7 +273,12 @@ static int get_pty_on_host(struct lxc_container *c, struct wrapargs *wrap, int *
 	if (lxc_console_create(conf) < 0)
 		return -1;
 	ts = conf->console.tty_state;
-	ts->stdoutfd = STDOUT_FILENO;
+	/* If stdout is not a tty but stderr is, we write to stderr. */
+	if (!isatty(STDOUT_FILENO) && isatty(STDERR_FILENO))
+		ts->stdoutfd = STDERR_FILENO;
+	else
+		ts->stdoutfd = STDOUT_FILENO;
+
```

Comment:
The comment and intent is highly unclear -- why would we *ever* redirect the container command's stdout to stderr? It shouldn't matter what input redirection is being done on the host. This looks like very unexpected and untypical Unix behaviour and magic?